### PR TITLE
✨ Enable native Custom Elements v1

### PR DESCRIPTION
**Pull request by jridgewell**
This will attempt to use native CEv1 when supported.

Based on whether the anonymous class remains a real, native ES6 class, we can use 0 polyfill (fully native support). If it's transpiled into a `function () {}`, then we can use a minimal polyfill that wraps the `HTMLElement` constructor only.

### CODE REVIEW

File: test/unit/test-amp-img.js [hunk 1]
```diff
@@ -377,6 +377,8 @@ describe('amp-img', () => {
   it('should propagate the object-fit attribute', () => {
     return getImg({
       src: '/examples/img/sample.jpg',
```
Comment (cathyxz) [inline test/unit/test-amp-img.js]: Were these test breaking due to this change (if so why?)?

Comment (jridgewell) [reply test/unit/test-amp-img.js]: I think the native Custom Elements implementations are able to fire `connectedCallback` much quicker than the `MutationObserver` the polyfill uses.

**Comment by jridgewell** [comment]
I created a new RTV experiment so that this can be tested on a smaller population.

/cc @lannka and @zhouyx, I also updated the `.eslintrc` so that RTV experiments are recognized as globals. It'd be really convenient if we could change to a `.eslintrc.js` file, so that we could dynamically generate the defined experiment globals from `experiments-config.json`. @rsimha?

**Comment by zhouyx** [comment]
> I also updated the .eslintrc so that RTV experiments are recognized as globals. It'd be really convenient if we could change to a .eslintrc.js file, so that we could dynamically generate the defined experiment globals from experiments-config.json. @rsimha?

Sounds like a great idea!

File: .eslintrc [hunk 2]
```diff
@@ -20,9 +20,13 @@
     "sourceType": "module"
   },
   "globals": {
-    "AMP": false,
-    "context": false,
-    "global": false
+    "ANALYTICS_VENDOR_SPLIT": "readonly",
+    "NATIVE_CUSTOM_ELEMENTS_V1": "readonly",
+    "_RTVEXP_INABOX_LITE": "readonly",
```
Comment (rsimha) [inline .eslintrc]: Can you move these to `build-system/global-configs/.eslintrc`?

Comment (jridgewell) [reply .eslintrc]: I don't know what that is.

Comment (rsimha) [reply .eslintrc]: `.eslintrc` files are hierarchical. Just add a new one under `build-system/global-configs/`

image

Comment (jridgewell) [reply .eslintrc]:
```
src/friendly-iframe-embed.js
  862:5  error  'NATIVE_CUSTOM_ELEMENTS_V1' is not defined  no-undef
```

I'm gonna leave this for you to do.

Comment (rsimha) [reply .eslintrc]: Oh, I see. The strings are used in actual code as well.

> I'm gonna leave this for you to do.

SGTM.

**Pull request merged**